frontend/src/app/offers/create-offer/create-offer.component.ts

```diff
@@ -0,0 +1,192 @@
+    .do(offer => this.toDataUrl(offer.image, (base64) => this.offer.image = {
+      content: base64,
+      filename: 'image.jpg'
+    }))
+    .subscribe((response: BaseOffer) => {
+      this.offer = response as AppOffer;
+      this.inEditMode = true;
+      this.offer = this.removeTime();
+      this.form.patchValue(this.offer);
+    });
+  }
+
+// Temporary fix to make editing work properly, to delete after we fix datetime issue
+  removeTime(){
+    this.offer.startedAt = this.offer.startedAt.replace(/T[0-9][0-9]:[0-9][0-9]:[0-9][0-9]Z$/, '');
+    this.offer.finishedAt = this.offer.finishedAt.replace(/T[0-9][0-9]:[0-9][0-9]:[0-9][0-9]Z$/, '');
+    if(this.offer.recruitmentEndDate){
+      this.offer.recruitmentEndDate = this.offer.recruitmentEndDate.replace(/T[0-9][0-9]:[0-9][0-9]:[0-9][0-9]Z$/, '');
+    };
+    if(this.offer.recruitmentStartDate){
+      this.offer.recruitmentStartDate = this.offer.recruitmentStartDate.replace(/T[0-9][0-9]:[0-9][0-9]:[0-9][0-9]Z$/, '');
+    };
+    if(this.offer.reserveRecruitmentEndDate){
+      this.offer.reserveRecruitmentEndDate = this.offer.reserveRecruitmentEndDate.replace(/T[0-9][0-9]:[0-9][0-9]:[0-9][0-9]Z$/, '');
+    }
+    if(this.offer.reserveRecruitmentStartDate){
+      this.offer.reserveRecruitmentStartDate = this.offer.reserveRecruitmentStartDate.replace(/T[0-9][0-9]:[0-9][0-9]:[0-9][0-9]Z$/, '');
+    }
+    return this.offer;
+  }
+
+  // TODO - delete when we decide what date format we want to have
+  addTime(){
+    this.form.value.startedAt = this.form.value.startedAt + 'T00:00:00Z';
+    this.form.value.finishedAt = this.form.value.finishedAt + 'T00:00:00Z';
+
+    if (this.form.value.recruitmentEndDate) {
+      this.form.value.recruitmentEndDate = this.form.value.recruitmentEndDate + 'T00:00:00Z';
+    }
+    if (this.form.value.recruitmentStartDate) {
+      this.form.value.recruitmentStartDate = this.form.value.recruitmentStartDate + 'T00:00:00Z';
+    }
+    if (this.form.value.reserveRecruitmentEndDate) {
+      this.form.value.reserveRecruitmentEndDate = this.form.value.reserveRecruitmentEndDate + 'T00:00:00Z';
+    }
+    if (this.form.value.reserveRecruitmentStartDate) {
+      this.form.value.reserveRecruitmentStartDate = this.form.value.reserveRecruitmentStartDate + 'T00:00:00Z';
+    }
+  }
+
+  ngOnDestroy() {
+    if (this.userSubscription) {
+      this.userSubscription.unsubscribe();
+    }
+  }
+
+  toDataUrl(url, callback) {
+    const xhr = new XMLHttpRequest();
+    xhr.onload = function() {
+        const reader = new FileReader();
+        reader.onloadend = function() {
+            callback(reader.result);
+        }
+        reader.readAsDataURL(xhr.response);
+    };
+    xhr.open('GET', url);
+    xhr.responseType = 'blob';
+    xhr.send();
+}
+
+  onSubmit(offer: AppOffer) {
+    if(this.form.valid){
+      this.addTime()
+    }
+    if(this.offer.image){
+      this.form.value.image = this.offer.image;
```

Comment:
This line in conjunction with the next line introduces a bug, that will result in 400 later.
`this.form.value.image = this.offer.image`
Writing it like this means that both form image and offer image are pointing to the same object, so in the next line you modify the content of form image and offer image in the same time. 

We should clone this.offer.image instead, and assign the clone to the form.
This should work:
`this.form.value.image = Object.assign({}, this.offer.image);`

Object.assign({}, object) assigns all the properties from object to the new {} object. It works only as a shallow copy and thats what we need here. If the object would contain another objects inside it would not work, as nested objects still would not be cloned.

Update:
As it turned out, TypeScript supports object spread operator (JavaScript is preparing for that, but not supporting it yet) so we can write it in a cleaner way:
`this.form.value.image = { ...this.offer.image };`

The spread operator `...` spreads all the properties form this.offer.image into the new {} object - as a result we get a shallow copy of the object as well.